api/types/okta.go

```diff
@@ -396,6 +396,79 @@ func (o *OktaAssignmentV1) CheckAndSetDefaults() error {
 	return nil
 }
 
+// IsEqual determines if two okta assignment resources are equivalent to one another.
+func (o *OktaAssignmentV1) IsEqual(i interface{}) bool {
+	ptr, ok := i.(*OktaAssignmentV1)
+	if ok {
+		if o == nil && ptr == nil {
+			return true
+		}
+		if o != nil && ptr != nil {
+			return o.isEqual(*ptr)
+		}
+		return false
+	}
+
+	other, ok := i.(OktaAssignmentV1)
+	if ok {
+		return o.isEqual(other)
+	}
+
+	return false
+
+}
+
+func (o OktaAssignmentV1) isEqual(other OktaAssignmentV1) bool {
```

Comment:
I wonder if we need to leverage something like https://www.jvt.me/posts/2023/03/27/go-generate-equal-goderive/ to reduce the cruft that's going to be associated with maintaining these IsEqual funcs 😓

Reply:
So I think `goderive` is the right way to go here. However, because these are all `proto` messages, it generates equivalence on the `XXX_` prefixed fields:

```
			this.XXX_NoUnkeyedLiteral == that.XXX_NoUnkeyedLiteral &&
			bytes.Equal(this.XXX_unrecognized, that.XXX_unrecognized) &&
			this.XXX_sizecache == that.XXX_sizecache
```

I think we could leverage this, but I think we'll need to do some customization beforehand. I'm happy to do this, but this PR is a bit more urgent than that. Any objection to taking this as a follow on?

Reply:
Happy to have this is a follow on from my point of view - my fear is if we leave it like this too long, we're going to end up maintaining thousands of lines of code and tests for IsEqual.

Reply:
I actually went ahead and did this. Let me know what you think.